docs/reference/search/retriever.asciidoc

```diff
@@ -0,0 +1,225 @@
+retriever with two or more children is considered a compound retriever. This
+allows for complex behavior to be depicted in a tree-like structure, called
+the retriever tree, to better clarify the order of operations that occur
+during a search.
+
+The following retrievers are available:
+
+`standard`::
+A <<standard-retriever, retriever>> that replaces the functionality of a traditional <<query-dsl, query>>.
+
+`knn`::
+A <<knn-retriever, retriever>> that replaces the functionality of a <<search-api-knn, knn search>>.
+
+`rrf`::
+A <<rrf-retriever, retriever>> that produces top documents from <<rrf, reciprocal rank fusion (RRF)>>.
+
+[[standard-retriever]]
+==== Standard Retriever
+
+A standard retriever returns top documents from a traditional <<query-dsl, query>>.
+
+===== Parameters:
+
+`query`::
+(Optional, <<query-dsl, query object>>)
++
+Defines a query to retrieve a set of top documents.
+
+`filter`::
+(Optional, <<query-dsl, query object or list of query objects>>)
++
+Applies a <<query-dsl-bool-query, boolean query filter>> to this retriever
+where all documents must match this query but do not contribute to the score.
+
+`search_after`::
+(Optional, <<search-after, search after object>>)
++
+Defines a search after object parameter used for pagination.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=terminate_after]
+
+`sort`::
++
+(Optional, <<sort-search-results, sort object>>)
+A sort object that that specifies the order of matching documents.
+
+`min_score`::
+(Optional, `float`)
++
+Minimum <<relevance-scores, `_score`>> for matching documents. Documents with a
+lower `_score` are not included in the top documents.
+
+`collapse`::
+(Optional, <<collapse-search-results, collapse object>>)
++
+Collapses the top documents by a specified key into a single top document per key.
+
+===== Restrictions
+
+When a retriever tree contains a compound retriever (a retriever with two or more child
+retrievers) *only* the query element is allowed.
+
+===== Example
+
+[source,js]
+----
+GET /index/_search
+{
+    "retriever": {
+        "standard": {
+            "query" { ... },
+            "filter" { ... },
+            "min_score": ...
+        }
+    },
+    "size": ...
+}
+----
+// NOTCONSOLE
+
+[[knn-retriever]]
+==== kNN Retriever
+
+A kNN retriever returns top documents from a <<knn-search, k-nearest neighbor search (kNN)>>.
+
+===== Parameters
+
+`field`::
+(Required, string)
++
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=knn-field]
+
+`query_vector`::
+(Required if `query_vector_builder` is not defined, array of `float`)
++
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=knn-query-vector]
+
+`query_vector_builder`::
+(Required if `query_vector` is not defined, query vector builder object)
++
+Defines a <<knn-semantic-search, model>> to build a query vector.
+
+`k`::
+(Required, integer)
++
+Number of nearest neighbors to return as top hits. This value must be fewer than
```

Comment:
Is it fewer or <=?

Reply:
Good catch! I added an equal to as well.